akka-distributed-data/src/main/scala/akka/cluster/ddata/protobuf/ReplicatedDataSerializer.scala

```diff
@@ -536,9 +573,131 @@ class ReplicatedDataSerializer(val system: ExtendedActorSystem)
       entries)
   }
 
+  def singleMapEntryFromProto[PEntry <: GeneratedMessage, A <: GeneratedMessage, B <: ReplicatedData](entry: PEntry, valueCreator: A ⇒ B)(implicit eh: ProtoMapEntryReader[PEntry, A]): Map[Any, B] = {
+    val elem = if (eh.hasStringKey(entry)) Some(eh.getStringKey(entry) → valueCreator(eh.getValue(entry)))
+    else if (eh.hasIntKey(entry)) Some(eh.getIntKey(entry) → valueCreator(eh.getValue(entry)))
+    else if (eh.hasLongKey(entry)) Some(eh.getLongKey(entry) → valueCreator(eh.getValue(entry)))
+    else if (eh.hasOtherKey(entry)) Some(otherMessageFromProto(eh.getOtherKey(entry)) → valueCreator(eh.getValue(entry)))
+    else None
+    if (elem.isDefined)
```

Comment:
`Option.get` is not idiomatic Scala (it's a bad smell, because it must almost be accompanied with `.isDefined`.

Most idiomatic is `.getOrElse`, but we avoid that because it's allocating a function instance (yeah, we are at an abnormal level with regards to performance concerns). In Scala 2.12 I think that is not an issue any more.

Therefore we prefer `match` on `Option`.